File: backend/models/Unit.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const UnitSchema = new Schema({
  name: {
    type: String,
    required: true,
    trim: true,
    index: true
  },
  propertyId: {
    type: Schema.Types.ObjectId,
    ref: 'Property',
    required: true,
    index: true
  },
  deviceIds: [{
    type: Schema.Types.ObjectId,
    ref: 'Device',
    index: true
  }],
  tenantUserIds: [{
    type: Schema.Types.ObjectId,
    ref: 'User',
    index: true
  }],
  commonAreaAccessIds: [{
    type: Schema.Types.ObjectId,
    ref: 'Device', // Assuming these are IDs of devices in common areas
    index: true
  }]
}, { timestamps: true });

module.exports = mongoose.model('Unit', UnitSchema); 

File: Sources/Models/Unit.swift
import Foundation

public struct Unit: Codable, Identifiable, Hashable {
    public let id: String
    public var name: String                     // e.g., "Apartment 101", "Suite 2B"
    public let propertyId: String               // Link to the parent Property
    public var tenantUserIds: [String]          // IDs of users who are tenants of this unit
    public var deviceIds: [String]              // IDs of devices (like locks) within this unit
    public var commonAreaAccessIds: [String]?   // IDs of common areas this unit has access to (e.g. main entrance lock ID)
    public var createdAt: Date?
    public var updatedAt: Date?

    public init(
        id: String,
        name: String,
        propertyId: String,
        tenantUserIds: [String] = [],
        deviceIds: [String] = [],
        commonAreaAccessIds: [String]? = nil,
        createdAt: Date? = Date(),
        updatedAt: Date? = Date()
    ) {
        self.id = id
        self.name = name
        self.propertyId = propertyId
        self.tenantUserIds = tenantUserIds
        self.deviceIds = deviceIds
        self.commonAreaAccessIds = commonAreaAccessIds
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
} 